sdk/documenttranslation/Azure.AI.DocumentTranslation/tests/DocumentTranslationClientTests.cs

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using Azure.Core;
+using Azure.Core.TestFramework;
+using Azure.Identity;
+using NUnit.Framework;
+
+namespace Azure.AI.DocumentTranslation.Tests
+{
+    public class DocumentTranslationClientTests : ClientTestBase
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DocumentTranslationClientTests"/> class.
+        /// </summary>
+        /// <param name="isAsync">A flag used by the Azure Core Test Framework to differentiate between tests for asynchronous and synchronous methods.</param>
+        public DocumentTranslationClientTests(bool isAsync)
+            : base(isAsync)
+        {
+        }
+
+        #region client
+        /// <summary>
+        /// Verifies functionality of the <see cref="DocumentTranslationClient"/> constructors.
+        /// </summary>
+        [Test]
+        public void ConstructorRequiresTheEndpoint()
+        {
+            var tokenCredential = new DefaultAzureCredential();
+            var keyCredential = new AzureKeyCredential("key");
+
+            Assert.Throws<ArgumentNullException>(() => new DocumentTranslationClient(null, tokenCredential));
```

Comment:
Nit: this is easier to add a bunch of test cases, but it's easier to identify exactly what failed if you keep 1 subject-under-test (SUT) per test (the extreme is one assertion, but that's pretty extreme).

That said, these all seem to be pretty unlikely to fail a single assertion in this bulk-style testing throughout this file. Just thought I'd bring it up.

Reply:
I like the idea of having modularized tests, but how concerned we are about increasing the size of the repo with too many recordings for live testing? In Metrics Advisor, we opted for testing different scenarios in a single test to avoid too many repeated requests. Things got a bit messy because of this, though.

I wonder if this could be avoided with good global setup methods.

Reply:
For parameter validation I think bulk is fine.
I agree with you that for the rest of the tests we should try to make it as specific as possible. I will keep that in mind here and if it starts exploding reconsider